proposals/0031-astro-watch-check.md

```diff
@@ -0,0 +1,92 @@
+<!--
+  Note: You are probably looking for `stage-1--discussion-template.md`!
+  This template is reserved for anyone championing an already-approved proposal.
+
+  Community members who would like to propose an idea or feature should begin
+  by creating a GitHub Discussion. See the repo README.md for more info.
+
+  To use this template: create a new, empty file in the repo under `proposals/${ID}.md`.
+  Replace `${ID}` with the official accepted proposal ID, found in the GitHub Issue
+  of the accepted proposal.
+-->
+
+- Start Date: 2023-02-27
+- Reference Issues: https://github.com/withastro/roadmap/issues/473
+- Implementation PR: https://github.com/withastro/astro/pull/6356
+
+# Summary
+
+Add a `--watch` flag to the `astro check` command.
+
+# Background & Motivation
+
+From the relevant issue:
+
+> In development mode there isn't a way to know if your .astro files contain no errors,
+> aside from editor feedback.
+>
+> If you use astro check you will know about errors, but might not see them until CI unless you remember to run the command before pushing.
+
+# Goals
+
+- A way to check TS errors in .astro files during development workflow.
+
+# Non-Goals
+
+- Having errors reported in the UI. This could be a nice enhancement in the future but would come with some downsides (such as possibly slowing down dev) that we want to punt on.
+
+# Detailed Design
+
+The suggested solution will use [`chokidar`](https://www.npmjs.com/package/chokidar), a battle tested
+file watcher for Node.js.
+
+The file watcher will ignore files inside `node_modules` by default and listen to changes to `.astro` files.
+
+The output emitted will be the same, compared to `astro check`.
+
+```shell
+astro check --watch
+```
+
+Will output:
+
+```block
+13:46:46 [check] Checking files in watch mode
+```
+
+The rest of the output will be the same as the command `astro check`.
+
+# Testing Strategy
+
+Override the default logger with one that is possible to inspect, listen to the stream of messages emitted by the watcher
+and make sure that the messages received are correct.
+
+Although, these kinds of tests are very brittle and unstable, because they require
+listening to the output of a command, and this command might take a different amount of time based on the OS.
```

Comment:
I think this isn't true now? This paragraph can be removed since the tests aren't listening to a command.